solutions/src/main/java/ru/sberbank/school/task02/exception/FxClientControllerImpl.java

```diff
@@ -0,0 +1,93 @@
+package ru.sberbank.school.task02.exception;
+
+import lombok.NonNull;
+import ru.sberbank.school.task02.ExternalQuotesService;
+import ru.sberbank.school.task02.FxClientController;
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.FxRequest;
+import ru.sberbank.school.task02.util.FxResponse;
+import ru.sberbank.school.task02.util.Symbol;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FxClientControllerImpl implements FxClientController {
+
+    private FxConversionServiceImpl fxConversionServiceImpl;
+
+    public FxClientControllerImpl(@NonNull ExternalQuotesService externalQuotesService) {
+        fxConversionServiceImpl = new FxConversionServiceImpl(externalQuotesService);
+    }
+
+    @Override
+    public List<FxResponse> fetchResult(List<FxRequest> requests) {
+        List<FxResponse> responses = new ArrayList<>();
+        for (FxRequest request : requests) {
+            responses.add(fetchResult(request));
+        }
+        return responses;
+    }
+
+    @Override
+    public FxResponse fetchResult(FxRequest requests) {
+        FxResponse response;
+
+        Symbol symbol = getSymbol(requests.getSymbol());
+        ClientOperation clientOperation = getDirection(requests.getDirection());
+        BigDecimal amount = getAmount(requests.getAmount());
+        BigDecimal price = fxConversionServiceImpl.convert(clientOperation, symbol, amount);
+
+        if(price != null) {
```

Comment:
WhitespaceAround: &#39;if&#39; is not followed by whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a2058961502>*